lib/pure/os.nim

```diff
@@ -175,21 +175,45 @@ proc osLastError*(): OSErrorCode =
 {.pop.}
 
 when defined(windows):
+  var
+    vi: OSVERSIONINFO
+
+  vi.dwOSVersionInfoSize = cast[DWORD](vi.sizeof)
+  discard getVersionExW(addr(vi))
+  let windowsXPorLater*: bool = vi.dwMajorVersion > 5 or 
```

Comment:
Actually while you're at it, could you move this version detection into `winlean` so that other modules can use it?

There are many ways to do this:

* A `isWindowsXPOrLater` and `isWindows7OrLater` procedures.
* A `getWindowsVersion` procedure that returns a type that can be compared to enums like `WindowsXP`, `Windows7`, etc. **(This would be ideal)** :)